enum.h

```diff
@@ -1283,4 +1283,17 @@ BETTER_ENUMS_CONSTEXPR_ map<Enum, T> make_map(T (*f)(Enum))
 
 }
 
+#ifndef BETTER_ENUMS_DECLARE_STD_HASH
+#define BETTER_ENUMS_DECLARE_STD_HASH(type) \
+	namespace std { \
+		template <> struct hash<type> \
+		{ \
+			size_t operator()(const type &x) const \
+			{ \
+				return std::hash<size_t>()(x._to_index()); \
```

Comment:
Why use the index rather than the value? `_to_index` runs a loop internally, whereas `_to_integral` is just a cast. Is this deliberate?

Reply:
Yes indeed. I am going to update that to use `_to_integral`.